backend/src/utils/adminUtils.ts

```diff
@@ -0,0 +1,151 @@
+import { CustomObjectsApi } from '@kubernetes/client-node';
+import { KubeFastifyInstance } from '../types';
+import { getAdminGroups, getAllowedGroups, getGroup } from './groupsUtils';
+
+const SYSTEM_AUTHENTICATED = 'system:authenticated';
+
+export const getGroupsConfig = async (
+  fastify: KubeFastifyInstance,
+  customObjectApi: CustomObjectsApi,
+  username: string,
+): Promise<boolean> => {
+  try {
+    const adminGroups = getAdminGroups();
+    const adminGroupsList = adminGroups.split(',');
+
+    if (adminGroupsList.includes(SYSTEM_AUTHENTICATED)) {
+      throw new Error('It is not allowed to set system:authenticated or as admin group.');
+    } else {
+      return await checkUserInGroups(fastify, customObjectApi, adminGroupsList, username);
+    }
+  } catch (e) {
+    fastify.log.error(e.toString());
+    return false;
+  }
+};
+
+export const isUserAdmin = async (
+  fastify: KubeFastifyInstance,
+  username: string,
+  namespace: string,
+): Promise<boolean> => {
+  const isAdmin: boolean = await isUserClusterRole(fastify, username, namespace);
+  return isAdmin ? isAdmin : (await getGroupsConfig(fastify, fastify.kube.customObjectsApi, username));
```

Comment:
heh, i got the or operator before changing to ternary because I thought parts of the code using ternary, so I just wanted a more cohesive codebase, but I agree, I'm reverting back to that I like it more.

Reply:
And regarding the lint issue... weird, i always pass a local test just to spot this, not sure what happened o.O